orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepository.java

```diff
@@ -56,14 +57,14 @@
 
   void delete(@Nonnull ExecutionType type, @Nonnull String id);
 
-  @Nonnull Observable<Execution> retrieve(ExecutionType type);
+  @Nonnull Iterable<Execution> retrieve(ExecutionType type);
```

Comment:
would it make sense to have these return Collection instead of Iterable? I don't think we actually are streaming results out of the implementation, and maybe simplifies some of the transformations to streams?

Reply:
or even Stream<Execution> for that matter

Reply:
I wanted to keep `Iterable<Execution>` because I want to introduce cursor-based pagination into the SQL implementation. What I was thinking would be to have a `class ExecutionIterable : Iterable<Execution>` that would automatically handle going back out to the database to fetch the next page of results without us, the end users, having to build in that functionality.

I'll take a look at what it'd take to return a Stream interface and still be able to provide the above functionality.

Reply:
Iterable is more cross-language friendly too. Stream is really only useful in Java.